tests/e2e/manifests/verify_manifest_lists.go

```diff
@@ -0,0 +1,674 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+verify_manifest_lists.go
+
+This program contains tests for Docker schema 2, multi-arch manifest lists
+that kubeadm requires for every release.
+
+First, it tries to get the latest Kubernetes release tags from GitHub.
+The list of release tags is filter so that it doesn't contain versions
+that are either too old or are known to not support manifest lists.
+
+It then creates a map of required images and image tags by looking
+at the constants defined in the kubeadm source code for a release
+branch estimated from a release tag.
+
+Once the images and image tags are defined, the program starts downloading
+the manifest lists from GCR, but also the actual images and layers,
+while verifying their contents and sizes. The layer download can be
+skipped by toggling `downloadLayers` to false.
+
+Results are cached so that a certain "image:tag" doesn't have to be verified
+more than once.
+*/
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/version"
```

Comment:
is this vendored in this repo? I don't see it...

Reply:
the `verify_manifest_lists.sh` script only pulls the `version` package instead of the whole apimachinery repo.
i didn't want to do `go get` or `dep`, because it only needs a couple of files in reality.
if there are objections to this i can change it.